UniqueID.md

```diff
@@ -0,0 +1,227 @@
+    - [Expose the same ID to all sites](#expose-the-same-id-to-all-sites)
+    - [Expose a method to convert an ID to its corresponding `FileSystemHandle`](#expose-a-method-to-convert-an-id-to-its-corresponding-filesystemhandle)
+    - [Tie the salt to the lifetime of the current browsing session](#tie-the-salt-to-the-lifetime-of-the-current-browsing-session)
+    - [Expose a truly persistent ID which is stable even after clearing site data](#expose-a-truly-persistent-id-which-is-stable-even-after-clearing-site-data)
+- [Stakeholder Feedback / Opposition](#stakeholder-feedback--opposition)
+- [References & acknowledgements](#references--acknowledgements)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Introduction
+
+Currently, `FileSystemHandle` objects can be opaquely serialized by the browser
+to be stored as values in IndexedDB. But there is no way for a site to generate
+a string from script which is guaranteed to be uniquely identifying for the file
+referenced by the `FileSystemHandle`.
+
+Developers have complained that building rich experiences on top of the File
+System Access API is challenging due to the inability to uniquely identify (and
+index on) handles.
+
+## Goals & Use Cases
+
+* Support O(1) access to a FileSystemHandle stored in IndexedDB
+* Two colluding sites cannot “join” a user by comparing IDs server-side
+* IDs should be stable across browsing sessions, making it suitable to use as a
+  key for storing the corresponding FileSystemHandle in IndexedDB
+* IDs should not be stable after clearing browsing data. In other words, this
+  should not be a truly persistent, unclearable identifier
+* IDs should not be stable across normal browsing mode and private browsing
+  modes
+* No information should be gleaned from the ID. It is completely opaque to the
+  site
+* Avoid surprises by having the same semantics as `isSameEntry()`, namely:
+  `await a.isSameEntry(b) === (await a.getUniqueId() === await b.getUniqueId())`
+
+## Non-goals
+
+* Support obtaining a `FileSystemHandle` from its ID
+
+## Use Cases
+
+### O(1) access to handles in IndexedDB
+
+An application wants to quickly tell whether it’s already seen a file which was
+recently selected from the file picker. Currently, storing handles in IndexedDB
+requires creating an arbitrary key mapping to a list of handles. Identifying
+whether a given handle is already stored in IndexedDB requires iterating through
+this list and calling the `isSameEntry()` method on each handle - an O(n)
+operation which is unacceptable given that sites may have access to an unbounded
+number of handles. A `FileSystemHandle`’s unique ID can be used as a key for
+IndexedDB, allowing for constant-time access.
+
+Before:
+
+```javascript
+// IndexedDB structure
+// { "handles": [handle1, handle2, … ] }
+const key = await handle.getUniqueId();
+var handles = await get("handles");
+handles.push(handle);
+await set("handles", handles);
```

Comment:
I don't understand this example.  `key` is assigned but unused.

Reply:
Whoops, `getUniqueId()` isn't available in the "before" state. This was a copy-paste error that I've now fixed